api/src/components/publicationVersion/controller.ts

```diff
@@ -437,3 +459,223 @@ export const create = async (
+                }
+            }
+        });
+
+        if (requestControlEvents.length) {
+            return response.json(403, { message: 'You have already requested control over this publication version.' });
+        }
+
+        // create new 'REQUEST_CONTROL' event
+        const requestControlEvent = await eventService.create('REQUEST_CONTROL', {
+            requesterId: event.user.id,
+            publicationVersion: { id: publicationVersion.id, versionOf: publicationVersion.versionOf }
+        });
+
+        // notify current corresponding author about the request
+        await email.requestControl({
+            eventId: requestControlEvent.id,
+            publicationVersion: {
+                id: publicationVersion.id,
+                versionOf: publicationVersion.versionOf,
+                title: publicationVersion.title || '',
+                authorEmail: publicationVersion.user.email || ''
+            },
+            requesterName: `${event.user.firstName} ${event.user.lastName}`
+        });
+
+        return response.json(200, { message: 'Successfully requested control over this publication version.' });
+    } catch (err) {
+        console.log(err);
+
+        return response.json(500, { message: 'Unknown server error.' });
+    }
+};
+
+export const approveControlRequest = async (
+    event: I.AuthenticatedAPIRequest<I.ApproveControlRequestBody, undefined, I.ApproveControlRequestPathParams>
+): Promise<I.JSONResponse> => {
+    const { id: publicationId, version } = event.pathParameters;
+    const { approve, eventId } = event.body;
+    const isApproved = approve === 'true';
+
+    try {
+        // get publication version
+        const publicationVersion = await publicationVersionService.get(publicationId, version);
+
+        if (!publicationVersion) {
+            return response.json(404, {
+                message: 'Publication version not found.'
+            });
+        }
+
+        // check that user is the current corresponding author
+        if (event.user.id !== publicationVersion.createdBy) {
+            return response.json(403, { message: 'You are not allowed to approve this control request.' });
+        }
+
+        // check if there's a request control event for the given 'eventId'
+        const requestControlEvent = await eventService.get(eventId);
+
+        if (!requestControlEvent) {
+            return response.json(404, { message: 'Control request not found.' });
+        }
+
+        const requestControlEventData = requestControlEvent.data as I.RequestControlData;
+        const {
+            requesterId,
+            publicationVersion: { id: eventVersionId }
+        } = requestControlEventData;
+
+        // check that version id from the event matches the current version id
+        if (eventVersionId !== publicationVersion.id) {
+            return response.json(400, { message: "Invalid 'eventId' provided." });
```

Comment:
Not sure about that message either, because the publication version details were also successfully used to identify the version and the corresponding author up to this point. In reality, either the "publicationId", "version" or "eventId" can be wrong at this stage (this is an edge case of course, if someone tries to do the API call programatically). I've changed to something like: `"There's no control request for this publication version with the provided 'eventId'"`. Hopefully this is a bit clearer